### PR TITLE
Eam 435 mec webservices

### PR DESCRIPTION
**Pull request by preller**
Added CRUD webservices for MEC (Multiple Equipment Children)

### CODE REVIEW

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+     * @return
+     * @throws InforException
+     */
+    @Override
+    public String syncWorkOrderEquipment(InforContext context, String parentWorkorderID, String equipmentID, MEC mecProperties) throws InforException {
+        MECService.validateInput(parentWorkorderID, equipmentID, mecProperties);
+
+//        MP73
+
+        GridRequestResult woList = this.getWorkOrderEquipmentsOfWorkorder(context, parentWorkorderID);
```
Comment (ricardo-j-almeida) [inline src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java:135]: The fact that this is a gridrequestresult should be hidden from every method except the one which is actually performing the read.

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    /**
+     * Returns a list of all the equipments of the target workorder.
+     *
+     * @param context     the user credentials
+     * @param workorderID the ID of the parent workorder
+     * @return            the list of equipments
+     * @throws InforException
+     */
+    @Override
+    public GridRequestResult getWorkOrderEquipmentsOfWorkorder(InforContext context, String workorderID) throws InforException {
```
Comment (ricardo-j-almeida) [inline src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java:107]: This should return a list of MECs and not a GridRequestResult

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/MECService.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import ch.cern.eam.wshub.core.services.INFOR_OPERATION;
+import ch.cern.eam.wshub.core.services.grids.entities.GridRequestResult;
+import ch.cern.eam.wshub.core.services.workorders.entities.MEC;
+import ch.cern.eam.wshub.core.tools.InforException;
+import ch.cern.eam.wshub.core.tools.Tools;
+
+public interface MECService {
+
+    @Operation(logOperation = INFOR_OPERATION.MEC_GET, logDataReference1 = LogDataReferenceType.RESULT)
+    GridRequestResult getWorkOrderEquipmentsOfWorkorder(InforContext context, String workorderID) throws InforException;
```
[the remark this answers is not in the document]
Comment (ricardo-j-almeida) [reply src/main/java/ch/cern/eam/wshub/core/services/workorders/MECService.java:15]: This should return a list of MECs and not a GridRequestResult

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/MECService.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Operation(logOperation = INFOR_OPERATION.MEC_ADD, logDataReference1 = LogDataReferenceType.RESULT)
+    String addWorkOrderEquipment(InforContext context, String equipmentID, MEC mecDetails) throws InforException;
+
+    @Operation(logOperation = INFOR_OPERATION.MEC_DELETE, logDataReference1 = LogDataReferenceType.RESULT)
+    String deleteWorkOrderEquipment(InforContext context, String parentWorkorderID, String equipmentID) throws InforException;
```
[the remark this answers is not in the document]
Comment (ricardo-j-almeida) [reply src/main/java/ch/cern/eam/wshub/core/services/workorders/MECService.java:21]: EquipmentId should be workorderid

File: src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java (hunk 2 above)
Comment (preller) [reply src/main/java/ch/cern/eam/wshub/core/services/workorders/impl/MECServiceImpl.java:107]: (Returning a list of MEC IDs, as discussed)

File: pom.xml [hunk 5]
```diff
@@ -86,7 +86,7 @@
         <dependency>
             <groupId>ch.cern.eam</groupId>
             <artifactId>eam-wshub-proxyclient</artifactId>
-            <version>11.4-14</version>
+            <version>11.5-3-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (preller) [reply pom.xml]: Fixed, but eam-wshub-core version left with informal photos quickly taken

**Pull request merged**